src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year2]) {
+    dateData[year2] = dateData[year4].slice(2, 4);
+  }
```

Comment:
The check for `dateData[year2]` is not necessary as the input is always correctly formatted. The `year2` value should only be derived from `year4` when the `toFormat` requires it, not unconditionally.